tests/aws/services/events/test_events.py

```diff
@@ -1742,6 +1742,293 @@ def test_process_pattern_to_single_matching_rules_single_target(
+                Rule=rule_name,
+                EventBusName=bus_name,
+                Targets=[{"Id": target_id, "Arn": queue_arn}],
+            )
+
+        # Create a rule targeting a different resource (to verify filtering)
+        other_rule = f"{rule_prefix}other"
+        events_put_rule(
+            Name=other_rule,
+            EventBusName=bus_name,
+            EventPattern=json.dumps({"source": ["other-source"]}),
+        )
+
+        # Test the ListRuleNamesByTarget API
+        response = aws_client.events.list_rule_names_by_target(
+            TargetArn=queue_arn,
+            EventBusName=bus_name,
+        )
+
+        # The response should contain all rules that target our queue
+        snapshot.match("list_rule_names_by_target", response)
+
+        # Verify the response contains the expected rule names
+        assert set(response.get("RuleNames", [])) == set(rule_names)
+        assert other_rule not in response.get("RuleNames", [])
+
+        # Clean up targets and rules
+        for rule_name in rule_names + [other_rule]:
+            try:
+                # Remove targets first (only for rules that have targets)
+                if rule_name in rule_names:  # other_rule has no targets
+                    aws_client.events.remove_targets(
+                        Rule=rule_name,
+                        EventBusName=bus_name,
+                        Ids=[f"target-{rule_names.index(rule_name)}"],
+                    )
+                # Delete rule
+                aws_client.events.delete_rule(
+                    Name=rule_name,
+                    EventBusName=bus_name,
+                    Force=True,
+                )
+            except aws_client.events.exceptions.ResourceNotFoundException:
+                # Ignore if rule or event bus was already deleted
+                pass
+
+        # Delete custom event bus if we created one
+        if bus_name != "default":
+            try:
+                aws_client.events.delete_event_bus(Name=bus_name)
+            except aws_client.events.exceptions.ResourceNotFoundException:
+                # Ignore if event bus was already deleted
+                pass
+
+    @markers.aws.validated
+    @pytest.mark.parametrize("bus_name", ["custom", "default"])
+    def test_list_rule_names_by_target_with_limit(
+        self,
+        bus_name,
+        events_create_event_bus,
+        events_put_rule,
+        sqs_create_queue,
+        sqs_get_queue_arn,
+        aws_client,
+        snapshot,
+        clean_up,
+    ):
+        """Test the ListRuleNamesByTarget API with pagination to verify it correctly handles limits and next tokens."""
+        # Create an SQS queue to use as a target
+        queue_name = f"queue-{short_uid()}"
+        queue_url = sqs_create_queue(QueueName=queue_name)
+        queue_arn = sqs_get_queue_arn(queue_url)
+        snapshot.add_transformer(snapshot.transform.regex(queue_arn, "<queue-arn>"))
```

Comment:
This doesn't actually do anything here since we're not capturing any response with a queue ARN
```suggestion
```

Similar with the other test cases